tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -422,6 +444,56 @@ class SPIRVEmitter : public ASTConsumer {
     }
   }
 
+  spv::Op translateOp(BinaryOperator::Opcode op, QualType type) {
+    // TODO: the following is not considering vector types yet.
+    const bool isSintType = type->isSignedIntegerType();
+    const bool isUintType = type->isUnsignedIntegerType();
+    const bool isFloatType = type->isFloatingType();
+
+#define BIN_OP_CASE_INT_FLOAT(kind, intBinOp, floatBinOp)                      \
+  \
+case BO_##kind : {                                                             \
+    if (isSintType || isUintType) {                                            \
+      return spv::Op::Op##intBinOp;                                            \
+    }                                                                          \
+    if (isFloatType) {                                                         \
+      return spv::Op::Op##floatBinOp;                                          \
+    }                                                                          \
+  }                                                                            \
+  break
+
+#define BIN_OP_CASE_SINT_UINT_FLOAT(kind, sintBinOp, uintBinOp, floatBinOp)    \
+  \
+case BO_##kind : {                                                             \
+    if (isSintType) {                                                          \
+      return spv::Op::Op##sintBinOp;                                           \
+    }                                                                          \
+    if (isUintType) {                                                          \
+      return spv::Op::Op##uintBinOp;                                           \
+    }                                                                          \
+    if (isFloatType) {                                                         \
+      return spv::Op::Op##floatBinOp;                                          \
+    }                                                                          \
+  }                                                                            \
+  break
+
+    switch (op) {
+      BIN_OP_CASE_INT_FLOAT(Add, IAdd, FAdd);
+      BIN_OP_CASE_INT_FLOAT(Sub, ISub, FSub);
+      BIN_OP_CASE_INT_FLOAT(Mul, IMul, FMul);
+      BIN_OP_CASE_SINT_UINT_FLOAT(Div, SDiv, UDiv, FDiv);
+      BIN_OP_CASE_SINT_UINT_FLOAT(Rem, SMod, UMod, FMod);
```

Comment:
Interestingly, I think the HLSL documentation for floating point % doesn't say enough to know if this should be FMod or FRem.  
https://msdn.microsoft.com/en-us/library/windows/desktop/bb509631(v=vs.85).aspx#Additive_and_Multiplicative_Operators 

FRem and FMod differ in where the result takes the sign from.   HLSL just says both sides have to have the same sign.  
I recommend that you add a comment here indicating this ambiguity, and that you're followng Glslang. Or even better,  check what Microsft's DXC compiler does!